v21.1/build-a-csharp-app-with-cockroachdb.md

```diff
@@ -36,168 +38,83 @@ Install the latest version of the [Npgsql driver](https://www.nuget.org/packages
-
-To convert the key to PKCS#12 format, run the following OpenSSL command on the `maxroach` user's key file in the directory where you stored your certificates:
-
-{% include copy-clipboard.html %}
-~~~ shell
-$ openssl pkcs12 -inkey client.maxroach.key -password pass: -in client.maxroach.crt -export -out client.maxroach.pfx
-~~~
-
-As of December 2018, you need to provide a password for this to work on macOS. See <https://github.com/dotnet/corefx/issues/24225>.
+## Step 4. Create a database
 
-## Step 6. Run the C# code
+{% include {{page.version.version}}/app/create-a-database.md %}
 
-Now that you have created a database and set up encryption keys, in this section you will:
-
-- [Create a table and insert some rows](#basic-example)
-- [Execute a batch of statements as a transaction](#transaction-example-with-retry-logic)
-
-### Basic example
-
-Replace the contents of `cockroachdb-test-app/Program.cs` with the following code:
-
-{% include copy-clipboard.html %}
-~~~ csharp
-{% include {{ page.version.version }}/app/basic-sample.cs %}
-~~~
+## Step 5. Run the C# code
 
-Then, run the code to connect as the `maxroach` user.  This time, execute a batch of statements as an atomic transaction to transfer funds from one account to another, where all included statements are either committed or aborted:
+Now that you have set up your project and created a database, you will run a basic application that creates a table and inserts some rows.
 
-{% include copy-clipboard.html %}
-~~~ shell
-$ dotnet run
-~~~
+### Get the code
 
-The output should be:
+Replace the contents of the `Program.cs` file that was automatically generated in your `cockroachdb-test-app` directory with the code below:
 
-~~~
-Initial balances:
-	account 1: 1000
-	account 2: 250
+{% include_cached copy-clipboard.html %}
+~~~ c#
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/hello-world-csharp/main/basic.cs %}
 ~~~
 
-### Transaction example (with retry logic)
+### Update the connection parameters
 
-Open `cockroachdb-test-app/Program.cs` again and replace the contents with the code shown below.
+<section class="filter-content" markdown="1" data-scope="local">
 
-{% include {{page.version.version}}/client-transaction-retry.md %}
+In a text editor, modify `Program.cs` with the settings to connect to the demo cluster:
 
-{% include copy-clipboard.html %}
+{% include_cached copy-clipboard.html %}
 ~~~ csharp
-{% include {{ page.version.version }}/app/txn-sample.cs %}
-~~~
-
-Then, run the code to connect as the `maxroach` user.  This time, execute a batch of statements as an atomic transaction to transfer funds from one account to another, where all included statements are either committed or aborted:
-
-{% include copy-clipboard.html %}
-~~~ shell
-$ dotnet run
-~~~
-
-The output should be:
-
-~~~
-Initial balances:
-	account 1: 1000
-	account 2: 250
-Final balances:
-	account 1: 900
-	account 2: 350
+connStringBuilder.Host = "{host-name}";
```

Comment:
This would already be `localhost` if we used different branches as I noted previously.